discover.go

```diff
@@ -30,6 +31,7 @@ type Provider interface {
 var Providers = map[string]Provider{
 	"aws":       &aws.Provider{},
 	"azure":     &azure.Provider{},
+	"azurevmss": &azurevmss.Provider{},
```

Comment:
Can you try to roll this into the existing `azure` provider since otherwise we're going to duplicate all dependencies. Maybe use a `type=vmss` parameter to make it unambiguous.